### ISSUE TITLE
5.5.9 - very high cpu usage, youtube tabs laggy and non-responsive

### ISSUE THREAD

**Issue by jamiew0w** [ajayyy/SponsorBlock#1988]
hello. i've noticed the last few days that youtube becomes extremely laggy if the same tab is used for several videos. it's been happening for a few days, i just haven't had time to look into it.
it becomes so laggy infact, that switching between tabs makes the youtube tab entirely black screen for a few seconds and small things such as pausing/starting a video can take several seconds.

initially i thought this was a ublock origin issue as youtube had rolled out changes a few months ago that caused the same behaviour but i had time today to test and:

1. windows, firefox, only sponsor block results in a laggy tab after ~60 minutes
2. windows, firefox, sponsor block disabled, other extensions enabled and everything is back to normal and quick
3. debian, firefox ESR, only sponsor block = same, laggy experience after ~60 minutes
4. debian, firefox ESR, sponsor block disabled (etc) and everything is good again.

there's nothing especially interesting in browser console, i could include things like screenshots of task manager etc, but it might be redundant? is there anything else i could provide? i'll update this issue with an output from firefox profiler once i get off work.
 sponsorblock worked wonderfully until very recently so i'm pretty certain it has to do with the 5.5.9 update.

additionally, are other people having the same issue?

**Comment by direraven1143** [comment]
similar issues for me on the 8th gen cpus in my house, particularly when watching lots of 4k or 60 fps video, but no issues on my main 12700kf/3080ti pc, using chrome on all pcs, disabling then re-enabling sponsorblock seems to fix the issue

**Comment by jamiew0w** [comment]
> similar issues for me on the 8th gen cpus in my house, particularly when watching lots of 4k or 60 fps video, but no issues on my main 12700kf/3080ti pc, using chrome on all pcs, disabling then re-enabling sponsorblock seems to fix the issue

I've found disabling/re-enabling seems to fix it too, which was a little frustrating because i was trying to replicate it and forward along what data i could. lol, go figure.
I'll leave this open for a bit - toggling isn't a massive hassle but it's not a fix either

**Comment by direraven1143** [comment]
I agree would love a fix that doesn't require toggling - I should also add that sponsorblock on my very underpowered $100 tablet seems to run 1080 HD youtube just fine without the same issue, so it seems to be older but still Windows 11 capable CPUs like Intel 8th gen where I seem to have an issue.

**Comment by direraven1143** [comment]
Any updates for this one or anything I can share to help? Now is happening on my gaming pc doing about 60 minutes 4k YouTube and toggling access for sponsorblock fixes the issue instantly?

**Comment by jamiew0w** [comment]
@ajayyy Might be good to ping you in here to see what information you need.

**Comment by MadeOfMagicAndWires** [comment]
I've been encountering a possibly related temporary slowdown for the last couple of days, youtube crawls to a halt when starting a video with the "This tab seems to be  slowing down your browser" popup, and after a few seconds it seems to get over itself and starts acting normally, I think after inserting segments into the video.

Purely anecdotally it might have  something to do with #1998? Both issues appeared around the second time for me.

**Comment by ajayyy** [comment]
@MadeOfMagicAndWires does this happen on specific videos?

**Comment by MadeOfMagicAndWires** [comment]
It seemed to affect pretty much any video I started, though I didn't test extensively. The issue has largely abated for me since the servers are back up, though the player UI still freezes up at times (sometimes with the firefox 'slowing down' warning).

I'll keep an eye out for when it happens to see if there's some kind of pattern, though because it's less consistent I don't know how reproducible it is. Not that I'm complaining.

**Comment by ajayyy** [comment]
Thanks, there's been some issues in the past with specific combinations of skips causing a code loop, so that's why I wanted to know if it is only on specific videos

**Comment by blindmodz** [comment]
July and is still an issue, watching a livestream was eating almost 7gb (single tab) and had to disable/enable SponsorBlock to stop this "leak" (couldn't replicate). Using Floorp with 10th Intel .

**Comment by herenti** [comment]
I am having extreme lag from sponsorblock on youtube. I am running firefox on linux

**Comment by molocko** [comment]
same

**Comment by andymerskin** [comment]
I've noticed it's far worse now since the latest YouTube UI update:
1. Hovering over the volume where it shows the slideout animation is laggy.
2. Pausing / playing videos has a significant delay after clicks.
3. Viewing thumbnails while hovering on the player track skips every few 100ms to catch up with the mouse cursor.
4. Typing comments shows significant delay between keystrokes and characters appearing on the screen, practically unusable because some characters get missed altogether, forcing you to go back and correct mistakes.

This is ruining YouTube for me, but SponsorBlock is an essential extension. Really hope this gets looked into sooner than later :(

**Comment by andymerskin** [comment]
I did some more digging and found Bitwarden's more recent update(s) to be a big culprit as well. For anyone reporting this issue here, if you're using Bitwarden, try disabling that extension and see if anything changes. If you profile a YouTube tab in Firefox's Task Manager, you'll notice Bitwarden is causing massive CPU spikes when interacting with various elements on YouTube. Their autofill service is doing something expensive in the background.
